src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs

```diff
@@ -49,15 +60,65 @@ public static string EmitVerb(Endpoint endpoint)
      * their validity (optionality), invoke the underlying handler with
      * the arguments bound from HTTP context, and write out the response.
      */
-    public static string EmitRequestHandler()
+    public static string EmitRequestHandler(this Endpoint endpoint)
     {
-        return """
-Task RequestHandler(HttpContext httpContext)
-                {
-                        var result = handler();
-                        return httpContext.Response.WriteAsync(result);
-                }
-""";
+        var code = new CodeWriter(new StringBuilder());
+        code.Indent(5);
+        code.WriteLine(endpoint.Response.IsAwaitable
+            ? "async Task RequestHandler(HttpContext httpContext)"
+            : "Task RequestHandler(HttpContext httpContext)");
+        code.StartBlock();
+
+        if (endpoint.Response.IsVoid)
+        {
+            code.WriteLine("handler();");
+            code.WriteLine("return Task.CompletedTask;");
+        }
+        else
+        {
+            code.WriteLine($"""httpContext.Response.ContentType ??= "{endpoint.Response.ContentType}";""");
+            if (endpoint.Response.IsAwaitable)
+            {
+                code.WriteLine("var result = await handler();");
+                code.WriteLine(endpoint.EmitResponseWritingCall());
+            }
+            else
+            {
+                code.WriteLine("var result = handler();");
+                code.WriteLine("return GeneratedRouteBuilderExtensionsCore.ExecuteObjectResult(result, httpContext);");
+            }
```

Comment:
`EmitResponseWritingCall` handles `IsAwaitable()`. So I think this can be refactored to just:

```suggestion
            code.WriteLine(endpoint.Response.IsAwaitable
                ? "var result = await handler();"
                : "var result = handler();");
            code.WriteLine(endpoint.EmitResponseWritingCall());
```